### PR TITLE
Change env/service variables --> variables

### PR DESCRIPTION
**Pull request by vipulgupta2048**
## Summary

Proposed change of how the shift to variables would look like. We can add more commits to this PR to get the change started. 
To be merged on 9th August for Renametheon.

## Things to do:

- [x]  Change further instances of serv/env variables
- [x]  Change screenshots (Check if all screenshots are correct, need to delete the ones that are extra)
- [x]  Change SDK command links (Not sure about Python SDK)
- [x]  Add redirects if needed to prevent pages from breaking
- [ ]  Take a look at masterclasses for changes to be done. Fix references here: https://github.com/balena-io/balena-fleet-management-masterclass
- [ ] Check if UI issues were solved - Specifically the one with Fleet wide define (because we need that ss)
- [ ]  Do a team ping when this is complete for others to make changes

﻿Signed-off-by: Vipul Gupta (@vipulgupta2048) <vipul@balena.io>

Thread: https://www.flowdock.com/app/rulemotion/i-docs/threads/EB5zr3eRfoY6bmJvc8DAeetZX48
Related PR: https://github.com/balena-io/balena-ui/pull/4839 and https://github.com/balena-io/balena-ui/pull/4889/files
Part of bigger work that is in #1969 (Not exactly linking to it)

---

> *Please make sure to read the [CONTRIBUTING](https://github.com/balena-io/docs/blob/master/CONTRIBUTING.md) document before opening the PR for relevant information on contributing to the documentation. Thanks!*

### CODE REVIEW

File: pages/learn/deploy/deploy-with-balena-button.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ You can further customize the behavior of the **Deploy with {{ $names.company.lo
 
 ### balena.yml configuration file
 
-Through the use of a `balena.yml` config file, you may also provide application [configuration](/learn/manage/configuration/) and [environment](/learn/manage/serv-vars/) variables. If provided, they are pre-populated in the advanced modal dialog when using the **Deploy with {{ $names.company.lower }}** button.
+Through the use of a `balena.yml` config file, you may also provide application [configuration](/learn/manage/configuration/) and [environment](/learn/manage/variables/) variables. If provided, they are pre-populated in the advanced modal dialog when using the **Deploy with {{ $names.company.lower }}** button.
```
Comment (chrisys) [inline pages/learn/deploy/deploy-with-balena-button.md:39]:
```suggestion
Through the use of a `balena.yml` config file, you may also specify [configuration](/learn/manage/configuration/) defaults and provide [variables](/learn/manage/variables/). If provided, these are pre-populated in the advanced modal dialog when using the **Deploy with {{ $names.company.lower }}** button.
```

File: pages/learn/getting-started.md [hunk 2]
```diff
@@ -185,7 +185,7 @@ Your application will then be downloaded and executed by all the devices you hav
 - Head to [balenaHub][balenahub], for more ready-to-deploy applications, or have a go at building and sharing your own cool projects using [balenaBlocks][balenablocks].
 - Try out [local mode][local-mode], which allows you to build and sync code to your device locally for rapid development.
 - Develop an application with [multiple containers][multicontainer] to provide a more modular approach to application management.
-- Manage your device fleet with the use of [configuration][configuration], [environment][service], and [service variables][service].
+- Manage your device fleet with the use of [configuration][configuration] and [environment variables][variables].
```
Comment (chrisys) [inline pages/learn/getting-started.md:188]:
```suggestion
- Manage your device fleet with the use of [configuration][configuration].
- Pass named values, parameters and data to your application using [variables][variables].
```

File: pages/learn/manage/variables.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+---
+title: Variables
+excerpt: Set dynamic-named values on the dashboard that affect the way running processes will behave on the device by using variables.
+---
+
+# Variables
+
+Variables allow you to provide runtime configuration to one or more running services without having to modify your source code. Variables can be added onto the device using the balenaCloud dashboard and can be assigned to either a specific service or all services running on the device. You can use variables to store secrets and other sensitive values out of your codebase and configure them when needed from the dashboard.
+ 
+Variables defined in the dashboard are added to the device as environement variables and can be accessed within the service that they were defined for as shown below:
```
Comment (chrisys) [inline pages/learn/manage/variables.md:10]:
```suggestion
Variables defined in the dashboard are exposed on the device as environment variables and can be accessed within the service or services that they were defined for as shown below:
```

File: pages/learn/manage/variables.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+---
+title: Variables
+excerpt: Set dynamic-named values on the dashboard that affect the way running processes will behave on the device by using variables.
```
Comment (chrisys) [inline pages/learn/manage/variables.md:3]:
```suggestion
excerpt: Use variables to pass named values to the device which can be used to affect the way running apps and processes will behave.
```

File: pages/learn/manage/variables.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+# Variables
+
+Variables allow you to provide runtime configuration to one or more running services without having to modify your source code. Variables can be added onto the device using the balenaCloud dashboard and can be assigned to either a specific service or all services running on the device. You can use variables to store secrets and other sensitive values out of your codebase and configure them when needed from the dashboard.
+ 
+Variables defined in the dashboard are added to the device as environement variables and can be accessed within the service that they were defined for as shown below:
+ 
+<img alt="Accessing variable" src="/img/variables/accessing_variables.png">
+
+__Note:__ Adding or modifying a variable for all services will restart all services on the device. Similarly for variables defined for specific services will only restart the affected service.
```
Comment (chrisys) [inline pages/learn/manage/variables.md:14]:
```suggestion
__Note:__ Adding or modifying a variable for all services will restart all services on the device. Similarly, changing variables defined only for a specific service will only restart the affected service.
```

File: pages/learn/manage/variables.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+__Note:__ Any variables defined through the dashboard do not apply to devices in [local mode][local-mode].
+
+Values can be up to 1MB (or approximately 1 million characters) in size each. A device will re-download the variables every time the state changes in the API, which may potentially result in a lot of [network traffic][bandwidth-control]. The interval can be configured with the [BALENA_SUPERVISOR_POLL_INTERVAL configuration variable][poll-interval].
+
+## Fleet-wide variables
+
+Variables defined at the fleet level are available to all devices in that fleet unless they are redefined with a variable of the same name at the device level.
+
+Fleet-wide variables can be found from the Fleet Summary page by clicking the *Variables* tabs.
```
Comment (chrisys) [inline pages/learn/manage/variables.md:28]:
```suggestion
Fleet-wide variables can be found from the Fleet Summary page by clicking the *Variables* tab.
```

File: pages/learn/manage/variables.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+To define a new variable, click the *Add variable* button in the upper-left corner.
+
+<img alt="Add variable" src="/img/variables/add_application_variable.png" width="80%">
+
+In the dialog box that opens, select either a specific service or all services to apply the variable too.
```
Comment (chrisys) [inline pages/learn/manage/variables.md:34]:
```suggestion
In the dialog box that opens, select either 'All services' or a specific service to which the variable will be applied.
```

**Comment by vipulgupta2048** [comment]
@balena-ci rebase

**Comment by vipulgupta2048** [comment]
@balena-ci rebase

**Comment by vipulgupta2048** [comment]
@balena-ci rebase

**Review by chrisys — APPROVED** [review]

**Pull request merged**